src/modules/place/PlaceView.js

```diff
@@ -0,0 +1,266 @@
+import React, {PropTypes, Component} from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Image,
+  Platform,
+  ActivityIndicator,
+  Dimensions,
+  ScrollView,
+  Linking
+} from 'react-native';
+import * as theme from '../../utils/theme';
+import * as Utils from '../../utils/utils';
+import Button from '../../components/Button';
+
+const window = Dimensions.get('window');
+
+class PlaceView extends Component {
+  static displayName = 'PlaceView';
+
+  static navigationOptions = {
+    title: 'Place',
+    header: {
+      tintColor: theme.colors.text,
+      style: {
+        backgroundColor: theme.colors.navBar
+      }
+    }
+  }
+
+  static propTypes = {
+    office: PropTypes.object.isRequired,
+    loading: PropTypes.bool.isRequired,
+    place: PropTypes.object.isRequired,
+    back: PropTypes.func.isRequired,
+    officeStateActions: PropTypes.shape({
+      retryPlace: PropTypes.func.isRequired
+    }).isRequired
+  };
+
+  componentWillMount() {
+    //if this.prop.place doesn't contain the right data go back to home screen
+    if (!this.props.place.name) {
+      this.props.back();
+    }
+  }
+
+  onNextPress = () => {
+    this.props.officeStateActions.retryPlace(this.props.office, this.props.place);
+  }
+
+  buildPhotosURL = () => {
+    var size = this.props.place.photos.groups[0].items.length;
+    var photo = this.props.place.photos.groups[0].items[Utils.getRamdonNumberBetweenRange(size, 0)];
+    return (photo.prefix + '500x500' + photo.suffix);
+  }
+
+  getImage = () => {
+    return (this.props.place.photos && this.props.place.photos.count && this.props.place.photos.count > 0)
+      ? (<Image style={styles.image} source={{uri: this.buildPhotosURL()}}>{this.getRating()}</Image>)
+      : (<View style={styles.noImage}>{this.getRating()}</View>);
+  }
+
+  getPrice = () => {
+    var price = '';
+    if (this.props.place.price) {
+      for (let i = 0; i < this.props.place.price.tier; i++) {
+        price += this.props.place.price.currency;
+      }
+    }
+    return price;
+  }
+
+  getCategories = () => {
+    var categories = '';
+    if (this.props.place.categories) {
+      for (let i = 0; i < this.props.place.categories.length; i++) {
+        categories += this.props.place.categories[i].name + ', ';
+      }
+    }
+    return categories.slice(0, -2);
```

Comment:
Again, very imperative code and we prefer using higher order functions and lodash as it usually is simpler and more error-prone as we're using hardened libraries. Plus this `slice`ing thing makes my head hurt, took me 5 minutes to realise it's cutting off the `, ` at the end...

```
const categories = this.props.place.categories || [];
const categoryNames = _.map(categories, 'name');
return _.join(categoryNames, ', ');
```